### PR TITLE
feat: adding softhsm2 to docker runner

### PR DESCRIPTION
**Pull request by NikolaMilosa**

### CODE REVIEW

File: docker/runner.Dockerfile [hunk 1]
```diff
@@ -87,6 +87,8 @@ USER runner
 WORKDIR /home/runner
 SHELL [ "/bin/bash", "-c" ]
 
+RUN source /opt/rye/env
```
Comment (sasa-tomic) [inline docker/runner.Dockerfile:90]: Can this go into a separate PR?

Comment (NikolaMilosa) [reply docker/runner.Dockerfile:90]: Okay, will wait both to be merged so I update only once, and not 2 times

**Review by LittleChimera — APPROVED** [review]
